jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -522,21 +522,16 @@ public void configureTempDirectory(File dir, WebAppContext context)
         if (dir == null)
             throw new IllegalArgumentException("Null temp dir");
 
-        //if dir exists and we don't want it persisted, delete it
-        if (dir.exists() && !context.isPersistTempDirectory())
+        // if it doesn't exist make it
+        if (!dir.exists())
```

Comment:
You've changed the logic here.   This was always deleting and recreating a non persistent temp dir.

We  don't want to do that any more because of hijacking!   But I think we have to ensure the tempdir is empty if it is not persistent.